org.alloytools.alloy.core/src/main/java/edu/mit/csail/sdg/alloy4/OurDialog.java

```diff
@@ -215,10 +227,14 @@ public static File askFile(boolean isOpen, String dir, final String ext, final S
 			open.setAlwaysOnTop(true);
 			open.setMode(isOpen ? FileDialog.LOAD : FileDialog.SAVE);
 			open.setDirectory(dir);
-			if (ext.length() > 0)
+			if (exts != null && exts.length > 0)
 				open.setFilenameFilter(new FilenameFilter() {
 					public boolean accept(File dir, String name) {
-						return name.toLowerCase(Locale.US).endsWith(ext);
+						for ( String ext : exts) {
```

Comment:
replace this `for` loop with something like `exts.toStream().any(ext -> endsWithIgnoreCase(name, ext))` (see my other comment regarding the helper method `endsWithIgnoreCase`

Reply:
Not opportune now, too many of those things in the code